phylum_project/src/lib.rs

```diff
@@ -93,7 +93,10 @@ pub fn find_project_conf(
     recurse_upwards: bool,
 ) -> Option<PathBuf> {
     let max_depth = if recurse_upwards { 32 } else { 1 };
-    let mut path = starting_directory.as_ref();
+    let path = starting_directory.as_ref();
+    // If `path` is like `.`, `path.parent()` is `None`.
+    let canonicalized = path.canonicalize();
+    let mut path = canonicalized.as_deref().ok().unwrap_or(path);
```

Comment:
I don't think there's any point to `unwrap_or` here? Seems like this would be much easier by just doing `canonicalize().ok()?`.

Reply:
`.ok()?` would cause the function to fail if the path does not exist, which the old code did not do.